### PR TITLE
HBASE-21714 Deprecated isTableAvailableWithSplit method in thrift module

### PR DESCRIPTION
**Pull request by nyl3532016**

### CODE REVIEW

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/ThriftHBaseServiceHandler.java [hunk 1]
```diff
@@ -702,13 +702,7 @@ public boolean isTableAvailable(TTableName tableName) throws TIOError, TExceptio
...
     }
   }
-
-  @Override
-  public boolean isTableAvailableWithSplit(TTableName tableName, List<ByteBuffer> splitKeys)
-      throws TIOError, TException {
-    throw new NotImplementedException("isTableAvailableWithSplit not supported");
-  }
-
+
```
Comment (Apache-HBase) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift2/ThriftHBaseServiceHandler.java]: whitespace:end of line

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 22 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 268 | master passed |
| +1 | compile | 35 | master passed |
| +1 | checkstyle | 34 | master passed |
| +1 | shadedjars | 271 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 82 | master passed |
| +1 | javadoc | 30 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 254 | the patch passed |
| +1 | compile | 32 | the patch passed |
| +1 | javac | 32 | the patch passed |
| -1 | checkstyle | 32 | hbase-thrift: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedjars | 270 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 528 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 89 | the patch passed |
| +1 | javadoc | 29 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 191 | hbase-thrift in the patch passed. |
| +1 | asflicense | 10 | The patch does not generate ASF License warnings. |
| | | 2267 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/201 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 670698c2f455 4.4.0-131-generic #157~14.04.1-Ubuntu SMP Fri Jul 13 08:53:17 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f9f6354393 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/1/artifact/out/diff-checkstyle-hbase-thrift.txt |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/1/artifact/out/whitespace-eol.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/1/testReport/ |
| Max. process+thread count | 2027 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 7 | https://github.com/apache/hbase/pull/201 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/201 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 308 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 335 | master passed |
| +1 | compile | 39 | master passed |
| +1 | checkstyle | 38 | master passed |
| +1 | shadedjars | 354 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 105 | master passed |
| +1 | javadoc | 39 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 326 | the patch passed |
| +1 | compile | 38 | the patch passed |
| +1 | javac | 38 | the patch passed |
| -1 | checkstyle | 42 | hbase-thrift: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 351 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 659 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 110 | the patch passed |
| +1 | javadoc | 35 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 313 | hbase-thrift in the patch passed. |
| +1 | asflicense | 12 | The patch does not generate ASF License warnings. |
| | | 3197 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/201 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d31631b5057f 4.4.0-145-generic #171-Ubuntu SMP Tue Mar 26 12:43:40 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f9f6354393 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/3/artifact/out/diff-checkstyle-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/3/testReport/ |
| Max. process+thread count | 1762 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 309 | master passed |
| +1 | compile | 39 | master passed |
| +1 | checkstyle | 39 | master passed |
| +1 | shadedjars | 345 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 103 | master passed |
| +1 | javadoc | 37 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 302 | the patch passed |
| +1 | compile | 37 | the patch passed |
| +1 | javac | 37 | the patch passed |
| +1 | checkstyle | 39 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 325 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 631 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 118 | the patch passed |
| +1 | javadoc | 39 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 360 | hbase-thrift in the patch passed. |
| +1 | asflicense | 13 | The patch does not generate ASF License warnings. |
| | | 2865 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/201 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 13ebe5c2c9e9 4.4.0-145-generic #171-Ubuntu SMP Tue Mar 26 12:43:40 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / f9f6354393 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/4/testReport/ |
| Max. process+thread count | 1758 (vs. ulimit of 10000) |
| modules | C: hbase-thrift U: hbase-thrift |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-201/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-thrift/src/main/resources/org/apache/hadoop/hbase/thrift2/hbase.thrift [hunk 2]
```diff
@@ -934,20 +934,6 @@ service THBaseService {
     1: required TTableName tableName
   ) throws (1: TIOError io)
 
-  /**
```
Comment (Apache9) [inline hbase-thrift/src/main/resources/org/apache/hadoop/hbase/thrift2/hbase.thrift]: I think we should deprecate it first before removing it?

**Pull request closed without merging**